### PR TITLE
Hoomd 3 examples

### PR DESCRIPTION
**Pull request by gustavor101**
The script examples for butane using Hoomd 3.x have been added. Each script follows the corresponding one in hoomd-blue folder.

### CODE REVIEW

File: hoomd3/classic/butane/ann/butane_ANN.py [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+    ax.set_xlabel(r"Dihedral Angle, $\xi$")
+    ax.set_ylabel(r"$A(\xi)$")
+
+    ax.plot(mesh, A)
+    plt.gca()
+    fig.savefig("butane-fe.png")
+
+    # write free energy to file
+    dih_vs_A = np.stack([mesh[:, 0], A[:, 0]], axis=1)
+    np.savetxt("butane-fe.dat", dih_vs_A, header='"Dihedral angle" "Free energy"')
```
Comment (ricalessandri) [inline hoomd3/classic/butane/ann/butane_ANN.py:313]: As discussed, let's switch to pickling (see this PR for an example: https://github.com/SSAGESLabs/PySAGES-examples/pull/14/files)

File: hoomd3/classic/butane/cff/butane_CFF.py [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+    method = CFF(cvs, grid, topology, kT, train_freq=args.train_freq)
+
+    timesteps = args.time_steps
+    tic = time.perf_counter()
+    run_result = pysages.run(method, generate_context, timesteps)
+    toc = time.perf_counter()
+    print(f"Completed the simulation in {toc - tic:0.4f} seconds.")
+
+    result = pysages.analyze(run_result)
+    return result["free_energy"]
```
Comment (ricalessandri) [inline hoomd3/classic/butane/cff/butane_CFF.py:298]: This script doesn't print to file the results. I think it's better to do so. And let's go for pickling (see this PR for an example: https://github.com/SSAGESLabs/PySAGES-examples/pull/14/files)

File: hoomd3/classic/butane/funn/butane_FUNN.py [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+    restraints = CVRestraints(lower=(-pi,), upper=(0,), kl=(10,), ku=(10,))
+
+    method = FUNN(cvs, grid, topology, train_freq=args.train_freq, restraints=restraints)
+
+    timesteps = args.time_steps
+
+    tic = time.perf_counter()
+    pysages.run(method, generate_context, timesteps)
+    toc = time.perf_counter()
+    print(f"Completed the simulation in {toc - tic:0.4f} seconds.")
```
Comment (ricalessandri) [inline hoomd3/classic/butane/funn/butane_FUNN.py:297]: Also this script is not printing out results, see the comment above for CFF (let's print something, and let's use pickling https://github.com/SSAGESLabs/PySAGES-examples/pull/14/files))

File: hoomd3/classic/butane/metad/butane_Metadynamics.py [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+
+    method = Metadynamics(cvs, height, sigma, stride, ngauss, deltaT=deltaT, kB=kB, grid=grid)
+
+    hills_file = "hills.dat"
+    callback = MetaDLogger(hills_file, stride) if args.log else None
+
+    tic = time.perf_counter()
+    pysages.run(method, generate_context, timesteps, callback)
+    toc = time.perf_counter()
+    print(f"Completed the simulation in {toc - tic:0.4f} seconds.")
```
Comment (ricalessandri) [inline hoomd3/classic/butane/metad/butane_Metadynamics.py:310]: Also this script is not printing out results (only `hills.dat`, which is only part of the results), see the comments above, incl. pickling https://github.com/SSAGESLabs/PySAGES-examples/pull/14/files

File: hoomd3/classic/butane/spectral_abf/butane_SpectralABF.py [hunk 5]
```diff
@@ -0,0 +1,321 @@
...
+    ax.set_xlabel(r"Dihedral Angle, $\xi$")
+    ax.set_ylabel(r"$A(\xi)$")
+
+    ax.plot(mesh, A)
+    plt.gca()
+    fig.savefig("butane-fe.png")
+
+    # write free energy to file
+    dih_vs_A = np.stack([mesh[:, 0], A[:, 0]], axis=1)
+    np.savetxt("butane-fe.dat", dih_vs_A, header='"dih", "Free energy"')
```
Comment (ricalessandri) [inline hoomd3/classic/butane/spectral_abf/butane_SpectralABF.py:314]: From .dat to pickle (https://github.com/SSAGESLabs/PySAGES-examples/pull/14/files)

**Review by ricalessandri — CHANGES_REQUESTED** [review]
Thanks a lot, @gustavor101 ! I've added a few comments mostly related homogenizing the way we output things (incl. the pickling that we already discussed).

File: hoomd3/classic/butane/ann/butane_ANN.py (hunk 1 above)
Comment (gustavor101) [reply hoomd3/classic/butane/ann/butane_ANN.py]: Done in [ANN] (https://github.com/SSAGESLabs/PySAGES-examples/pull/15/commits/8e853d9b1e0b49375e2e36d24c97c987a48baabe)

File: hoomd3/classic/butane/cff/butane_CFF.py (hunk 2 above)
Comment (gustavor101) [reply hoomd3/classic/butane/cff/butane_CFF.py:315]: Plot and pickling for CFF in https://github.com/SSAGESLabs/PySAGES-examples/pull/15/commits/6f6ce893013fd93698fb3637cbaf33eac2fbb4bf

File: hoomd3/classic/butane/funn/butane_FUNN.py (hunk 3 above)
Comment (gustavor101) [reply hoomd3/classic/butane/funn/butane_FUNN.py:299]: Modifications in https://github.com/SSAGESLabs/PySAGES-examples/pull/15/commits/2c31e05c289e1aaf8725b96b0d26ef8eebd59f82

File: hoomd3/classic/butane/metad/butane_Metadynamics.py (hunk 4 above)
Comment (gustavor101) [reply hoomd3/classic/butane/metad/butane_Metadynamics.py:312]: Plot pand pickle file in https://github.com/SSAGESLabs/PySAGES-examples/pull/15/commits/6da8457e8ad8671a45b0d8e0a078158159b39e9e

File: hoomd3/classic/butane/spectral_abf/butane_SpectralABF.py (hunk 5 above)
Comment (gustavor101) [reply hoomd3/classic/butane/spectral_abf/butane_SpectralABF.py]: Pickle was added in https://github.com/SSAGESLabs/PySAGES-examples/pull/15/commits/385c0aed6121b764767d117aeea5f6d8056d275e

**Review by ricalessandri — APPROVED** [review]
Looks good! Thanks a lot @gustavor101

**Pull request merged**